File: CSharp/Chess/Chess/Set.cs
using System;

namespace Chess
{
	public enum Color
	{
		Black,
		White
	}
	
	public class Set
	{	
		private King king;
		private Queen queen;
		private Rook rookA;
		private Rook rookH;
		private Knight knightB;
		private Knight knightG;
		private Bishop bishopC;
		private Bishop bishopF;
		private Pawn pawnA;
		private Pawn pawnB;
		private Pawn pawnC;
		private Pawn pawnD;
		private Pawn pawnE;
		private Pawn pawnF;
		private Pawn pawnG;
		private Pawn pawnH;
		
		private Color color;
		public Piece[] pieces;

		public Set(Color color)
		{
			int rank;
			this.color = color;
			pieces = new Piece[16];
		
			rank = (color == Color.Black) ? 8 : 1;
			pieces[2] = rookA = new Rook(color, new Position(rank, 'a'));
			pieces[6] = knightB = new Knight(color, new Position(rank, 'b'));
			pieces[4] = bishopC = new Bishop(color, new Position(rank, 'c'));
			pieces[0] = king = new King(color, new Position(rank, 'd'));
			pieces[1] = queen = new Queen(color, new Position(rank, 'e'));
			pieces[5] = bishopF = new Bishop(color, new Position(rank, 'f'));
			pieces[7] = knightG = new Knight(color, new Position(rank, 'g'));
			pieces[3] = rookH = new Rook(color, new Position(rank, 'h'));
			
			rank = (color == Color.Black) ? 7 : 2;
			pieces[8] = pawnA = new Pawn(color, new Position(rank, 'a'));
			pieces[9] = pawnB = new Pawn(color, new Position(rank, 'b'));
			pieces[10] = pawnC = new Pawn(color, new Position(rank, 'c'));
			pieces[11] = pawnD = new Pawn(color, new Position(rank, 'd'));
			pieces[12] = pawnE = new Pawn(color, new Position(rank, 'e'));
			pieces[13] = pawnF = new Pawn(color, new Position(rank, 'f'));
			pieces[14] = pawnG = new Pawn(color, new Position(rank, 'g'));
			pieces[15] = pawnH = new Pawn(color, new Position(rank, 'h'));
		}
	}
}

File: Ruby/Chess/set.rb

require './king'
require './queen'
require './rook'
require './knight'
require './bishop'
require './pawn'

class Set
  
  def initialize(color)
    @color = color
    @pieces = Array.new(16)
    
    rank = (color.eql? "Black") ? 8 : 1
    @pieces[0] = @rookA = Rook.new(@color, Position.new(rank, 'a'))
    @pieces[1] = @knightA = Knight.new(@color, Position.new(rank, 'b'))
    @pieces[2] = @bishopA = Bishop.new(@color, Position.new(rank, 'c'))
    @pieces[3] = @king = King.new(@color, Position.new(rank, 'd'))
    @pieces[4] = @queen = Queen.new(@color, Position.new(rank, 'e'))
    @pieces[5] = @bishopB = Bishop.new(@color, Position.new(rank, 'f'))
    @pieces[6] = @knightB = Knight.new(@color, Position.new(rank, 'g'))
    @pieces[7] = @rookB = Rook.new(@color, Position.new(rank, 'h'))
    
    rank = (color.eql? "Black") ? 7 : 2
    @pieces[8] = @pawnA = Pawn.new(@color, Position.new(rank, 'a'))
    @pieces[9] = @pawnB = Pawn.new(@color, Position.new(rank, 'b'))
    @pieces[10] = @pawnC = Pawn.new(@color, Position.new(rank, 'c'))
    @pieces[11] = @pawnD = Pawn.new(@color, Position.new(rank, 'd'))
    @pieces[12] = @pawnE = Pawn.new(@color, Position.new(rank, 'e'))
    @pieces[13] = @pawnF = Pawn.new(@color, Position.new(rank, 'f'))
    @pieces[14] = @pawnG = Pawn.new(@color, Position.new(rank, 'g'))
    @pieces[15] = @pawnH = Pawn.new(@color, Position.new(rank, 'h'))
  end
  
  def GetPieces()
    return @pieces
  end

end
